db/columns/utils.py

```diff
@@ -41,3 +41,24 @@ def init_mathesar_table_column_list_with_defaults(column_list):
             if c.name == constants.ID:
                 c.autoincrement = False
     return default_columns + given_columns
+
+
+def get_column_obj_from_relation(relation, column):
+    try:
+        column = find_column_by_name_in_relation(relation, column)
```

Comment:
Why is the `col_name_string` parameter being passed a `column`? `find_column_by_name_in_relation` seems to expect a string as that parameter.

Reply:
That's why it's in a `try` block. This function needs to deal with a `Column`, a `MathesarColumn`, or a (possibly namespaced) column name string.

Reply:
We had a conversation about this code, but it didn't really dawn on me until later that day when I was daydreaming that I would try to avoid having a routine whose input can be one of multiple different data structures (here that's `MathesarColumn`, SA Column or a column name string). I would consider the fact, that same data is being passed around in different forms, as the problem. Also, if a single routine accepts different forms of input, its number of states increases and it's less predictable (same goes for output, btw). I'd say that the simpler and more efficient we can make the data structures being passed around the better. Off topic, but reminds me of those programming maxims that say that solving a problem is about finding the data structure, not the algorithm. This might also relate to those conversations we've been having about duct typing and type annotations.